ksqldb-common/src/main/java/io/confluent/ksql/properties/with/CommonCreateConfigs.java

```diff
@@ -86,6 +87,14 @@ public static void addToConfigDef(
                 + "Kafka cluster configuration for replicas will be used for creating a new "
                 + "topic."
         )
+        .define(
+            SOURCE_TOPIC_RETENTION_IN_MS,
+            ConfigDef.Type.LONG,
+            null,
+            Importance.MEDIUM,
+            "The retention in milliseconds in the backing topic. If this property is"
+                + "not set then the default value of 7 days will be used for creating a new topic."
```

Comment:
> not set then the default value of 7 days will be used for creating a new topic.

the default for tables is infinite + compaction, which we should make sure that we maintain. Actually, we may want to consider not allowing users to specify this when they create a table altogether

Reply:
> Actually, we may want to consider not allowing users to specify this when they create a table altogether

I agree with this, @bvarghese1. Let's allow retention to be customized only for streams